react/features/conference/components/web/SalesforceLinkDialog.js

```diff
@@ -0,0 +1,374 @@
+ */
+function SalesforceLinkDialog() {
+    const { t } = useTranslation();
+    const classes = useStyles();
+    const dispatch = useDispatch();
+    const [ selectedRecord, setSelectedRecord ] = useState(null);
+    const [ selectedRecordOwner, setSelectedRecordOwner ] = useState(null);
+    const [ records, setRecords ] = useState([]);
+    const [ isLoading, setLoading ] = useState(false);
+    const [ searchTerm, setSearchTerm ] = useState(null);
+    const [ notes, setNotes ] = useState(null);
+    const [ hasRecordsErrors, setRecordsErrors ] = useState(false);
+    const [ hasDetailsErrors, setDetailsErrors ] = useState(false);
+    const conference = useSelector(getCurrentConference);
+    const sessionId = conference.getMeetingUniqueId();
+    const { salesforceUrl } = useSelector(state => state['features/base/config']);
+    const { jwt } = useSelector(state => state['features/base/jwt']);
+    const showSearchResults = searchTerm && searchTerm.length > 1;
+    const showNoResults = showSearchResults && records.length === 0;
+
+    useEffect(() => {
+        const fetchRecords = async () => {
+            setRecordsErrors(false);
+            setLoading(true);
+
+            try {
+                const text = showSearchResults ? searchTerm : null;
+                const result = await getSessionRecords(salesforceUrl, jwt, text);
+
+                setRecords(result);
+            } catch (error) {
+                logger.error('Encountered an error while fetching records.');
+                setRecordsErrors(true);
+            }
+
+            setLoading(false);
+        };
+
+        fetchRecords();
+    }, [ searchTerm, getSessionRecords, salesforceUrl, jwt ]);
+
+    useEffect(() => {
+        const fetchRecordDetails = async () => {
+            setDetailsErrors(false);
+            setSelectedRecordOwner(null);
+            try {
+                const result = await getSessionRecordDetails(salesforceUrl, jwt, selectedRecord);
+
+                setSelectedRecordOwner({
+                    id: result.id,
+                    name: result.ownerName,
+                    type: 'OWNER'
+                });
+            } catch (error) {
+                logger.error('Encountered an error while fetching record owner info.');
+                setDetailsErrors(true);
+            }
+        };
+
+        fetchRecordDetails();
+    }, [ selectedRecord, getSessionRecordDetails, salesforceUrl, jwt ]);
+
+
+    /**
+     * Callback for the onChange event of the search field.
+     */
+    const onChange = useCallback((event: Event) => {
+        const value = getFieldValue(event);
+
+        setSearchTerm(value);
+    }, [ getFieldValue ]);
+
+    /**
+     * Callback for the onDecline event of the dialog.
+     */
+    const resetRecord = useCallback(() => {
+        setSelectedRecord(null);
+    }, []);
+
+    /**
+     * Callback for the onSubmit event of the dialog.
+     */
+    const linkMeeting = useCallback(async () => {
```

Comment:
seems like there's a lot of code duplication between this component and the native one